src/routes/accounts/accounts.controller.ts

```diff
@@ -47,6 +51,22 @@ export class AccountsController {
     return this.accountsService.getDataTypes();
   }
 
+  @ApiOkResponse({ type: AccountDataSetting, isArray: true })
+  @Put(':address/data-settings')
+  @UseGuards(AuthGuard)
+  async upsertAccountDataSettings(
+    @Auth() authPayload: AuthPayload,
+    @Param('address', new ValidationPipe(AddressSchema)) address: `0x${string}`,
```

Comment:
We should rely on the authenticated `AuthPayload['address']` here. Otherwise, (as long as I'm authenticated) I could change the settings of any `address` here.

Reply:
As we agreed offline, we'll keep it as it is, as this address/auth alignment is been checked [here](https://github.com/safe-global/safe-client-gateway/blob/72449d149fa12382cd4c8c19416afbaaf519ddcc/src/domain/accounts/accounts.repository.ts).